js/ai/src/prompt.ts

```diff
@@ -222,3 +561,158 @@ export function isExecutablePrompt(obj: any): boolean {
     !!(obj as ExecutablePrompt)?.stream
   );
 }
+
+export async function loadPromptFolder(
+  registry: Registry,
+  dir: string = './prompts',
+  ns: string
+): Promise<void> {
+  const promptsPath = resolve(dir);
+  if (existsSync(promptsPath)) {
+    const dirEnts = readdirSync(promptsPath, {
+      withFileTypes: true,
+      recursive: true,
+    });
+    for (const dirEnt of dirEnts) {
+      if (dirEnt.isFile() && dirEnt.name.endsWith('.prompt')) {
+        if (dirEnt.name.startsWith('_')) {
+          const partialName = dirEnt.name.substring(1, dirEnt.name.length - 7);
+          definePartial(
+            registry,
+            partialName,
+            readFileSync(join(dirEnt.path, dirEnt.name), {
+              encoding: 'utf8',
+            })
+          );
+          logger.debug(
+            `Registered Dotprompt partial "${partialName}" from "${join(dirEnt.path, dirEnt.name)}"`
+          );
+        } else {
+          // If this prompt is in a subdirectory, we need to include that
+          // in the namespace to prevent naming conflicts.
+          let prefix = '';
+          if (promptsPath !== dirEnt.path) {
+            prefix = dirEnt.path
+              .replace(`${promptsPath}/`, '')
+              .replace(/\//g, '-');
+          }
+          await loadPrompt(registry, dirEnt.path, dirEnt.name, prefix, ns);
+        }
+      }
+    }
+  }
+}
+
+export function definePartial(
+  registry: Registry,
+  name: string,
+  source: string
+) {
+  registry.dotpromptEnv.definePartial(name, source);
+}
+
+export function defineHelper(
+  registry: Registry,
+  name: string,
+  fn: Handlebars.HelperDelegate
+) {
+  registry.dotpromptEnv.defineHelper(name, fn);
+}
+
+async function loadPrompt(
+  registry: Registry,
+  path: string,
+  filename: string,
+  prefix = '',
+  ns = 'dotprompt'
+): Promise<void> {
+  let name = `${prefix ? `${prefix}-` : ''}${basename(filename, '.prompt')}`;
+  let variant: string | null = null;
+  if (name.includes('.')) {
+    const parts = name.split('.');
+    name = parts[0];
+    variant = parts[1];
+  }
+  const source = readFileSync(join(path, filename), 'utf8');
+  const parsedPrompt = registry.dotpromptEnv.parse(source);
+  const promptMetadata =
+    await registry.dotpromptEnv.renderMetadata(parsedPrompt);
```

Comment:
Could we keep this around as-is somewhere in the registry metadata? David East is working on "extensions" of dotprompt and needs to be able to access the `ext` field of the rendered metadata.

Reply:
see line 660, is that enough?

Reply:
👍